src/scheduler.ts

```diff
@@ -190,13 +190,16 @@ export class Scheduler {
     const polkadotApi = await this.getAPIClient()
     const txObject = polkadotApi.tx(extrinsicHex)
     const unsub = await txObject.send(async (result) => {
+      const { status } = result;
       if (_.isNil(handleDispatch)) {
         await this.defaultErrorHandler(result)
       } else {
         await handleDispatch(result)
       }
+      if (status.isFinalized) {
+        unsub();
```

Comment:
hmmm can it call unsub() inside itself?

Reply:
Yes.
See polkadot.api examples.
https://polkadot.js.org/docs/api/start/api.tx.subs